code/GridFieldOrderableRows.php

```diff
@@ -450,10 +485,10 @@ protected function reorderItems($list, array $values, array $order) {
 			// *_versions table is updated. This ensures re-ordering works
 			// similar to the SiteTree where you change the position, and then
 			// you go into the record and publish it.
-			foreach(array_values($order) as $pos => $id) {
```

Comment:
In addition, array_values() forces the first $pos to be 0, which isn't good for checking if items have a sort set (=== 0 vs isset()).